### PR TITLE
Xsuaa support in cds-feature-identity

### PR DESCRIPTION
**Pull request by StefanHenke**

### CODE REVIEW

File: java/security.md [hunk 1]
```diff
@@ -41,48 +41,32 @@ Without security configured, CDS services are exposed to public. Proper configur
...
-2. The application is bound to an [XSUAA service instance](../guides/authorization#xsuaa-configuration)
-::: tip
-CAP Java picks only a single XSUAA binding. If you have multiple bindings, choose a specific binding with property `cds.security.xsuaa.binding`.
-Choose an appropriate XSUAA service plan to fit the requirements. For instance, if your service should be exposed as technical reuse service, make use of plan `broker`.
-:::
+### Configure XSUAA and IAS Authentication { #xsuaa-ias}
+To enable your application for XSUAA or IAS-authentication the following dependencies need to be set:
+   * `com.sap.cloud.security:resourceserver-security-spring-boot-starter` that brings [spring-security library](https://github.com/SAP/cloud-security-services-integration-library/tree/main/spring-security)
+   * `org.springframework.boot:spring-boot-starter-security` 
+   * `cds-feature-identity`
```
Comment (BraunMatthias) [inline java/security.md:50]: We shouild point to recommended cds-starteer-bundles cf and k8 here. We don't expect customers to set the dependecies individually.

File: java/security.md [hunk 2]
```diff
@@ -41,48 +41,32 @@ Without security configured, CDS services are exposed to public. Proper configur
...
 
 The individual dependencies can be explicitly added in the `pom.xml` file of your service.
 
-Recommended alternative is to use the `cds-starter-cloudfoundry` or the `cds-starter-k8s` starter bundle, which covers all required dependencies for XSUAA-authentication.
-
-### Configure IAS Authentication { #ias}
+In addition, your application needs to be bound to corresponding service instances depending on your scenario. If your application should:
```
Comment (BraunMatthias) [inline java/security.md:54]:
```suggestion
In addition, your application needs to be bound to corresponding service instances depending on your scenario. If your application should...
```

File: java/security.md [hunk 3]
```diff
@@ -41,48 +41,32 @@ Without security configured, CDS services are exposed to public. Proper configur
...
+   * only accept tokens issued by XSUAA, bind your application to a [XSUAA service instance](../guides/authorization#xsuaa-configuration)
```
Comment (BraunMatthias) [inline java/security.md:55]:
```suggestion
   * only accept tokens issued by XSUAA, bind your application to an [XSUAA service instance](../guides/authorization#xsuaa-configuration)
```

File: java/security.md [hunk 4]
```diff
@@ -41,48 +41,32 @@ Without security configured, CDS services are exposed to public. Proper configur
...
+   * only accept tokens issued by IAS, bind your application to a [IAS service instance](https://help.sap.com/docs/IDENTITY_AUTHENTICATION)
```
Comment (BraunMatthias) [inline java/security.md:56]:
```suggestion
   * only accept tokens issued by IAS, bind your application to an [IAS service instance](https://help.sap.com/docs/IDENTITY_AUTHENTICATION)
```

File: java/security.md [hunk 5]
```diff
@@ -41,48 +41,32 @@ Without security configured, CDS services are exposed to public. Proper configur
...
-
-It is possible to support IAS-authentication and XSUAA-authentication at the same time (hybrid). In this case, the CAP application will accept tokens issued by IAS and XSUAA.
-
-Your application is secured by the hybrid mode **automatically**, if
-1. You enabled IAS-authentication as described in [Configure IAS Authentication](#ias)
-2. The application is additionally bound to an [XSUAA service instance](../guides/authorization#xsuaa-configuration)
+### Transition from `cds-feature-xsuaa` to `cds-feature-identity`{ #transition-xsuaa-ias}
+CAP also provides support for XSUAA-based authentication via the maven dependency `cds-feature-xsuaa` which is based on the [spring-xsuaa library](https://github.com/SAP/cloud-security-services-integration-library/tree/main/spring-xsuaa). As the spring-xsuaa library is deprecated, it is recommended to move to `cds-feature-identity`.
 
+To do so, remove existing dependencies to `cds-feature-xsuaa` and `xsuaa-spring-boot-starter` and follow the description in [Configure XSUAA and IAS Authentication]{.#xsuaa-ias}.
```
Comment (BraunMatthias) [inline java/security.md:67]:
```suggestion
To do so, remove existing dependencies to `cds-feature-xsuaa` and `xsuaa-spring-boot-starter` and follow the description in [Configure XSUAA and IAS Authentication]{#xsuaa-ias}.  
```

**Review by BraunMatthias — COMMENTED** [review]
nice docu

File: java/security.md (hunk 2 above)
Comment (renejeglinsky) [reply java/security.md:54]:
```suggestion
In addition, your application needs to be bound to corresponding service instances depending on your scenario. The following list describes which services needs to be bound depending on the tokens your applications should accept:
```

File: java/security.md [hunk 6]
```diff
@@ -41,48 +41,32 @@ Without security configured, CDS services are exposed to public. Proper configur
...
 
 The individual dependencies can be explicitly added in the `pom.xml` file of your service.
 
-Recommended alternative is to use the `cds-starter-cloudfoundry` or the `cds-starter-k8s` starter bundle, which covers all required dependencies for XSUAA-authentication.
-
-### Configure IAS Authentication { #ias}
+In addition, your application needs to be bound to corresponding service instances depending on your scenario. If your application should:
+   * only accept tokens issued by XSUAA, bind your application to a [XSUAA service instance](../guides/authorization#xsuaa-configuration)
+   * only accept tokens issued by IAS, bind your application to a [IAS service instance](https://help.sap.com/docs/IDENTITY_AUTHENTICATION)
+   * accept tokens issued by XSUAA and IAS, bind your application to service instances of both types.
```
Comment (renejeglinsky) [inline java/security.md:57]:
```suggestion
   * accept tokens issued by XSUAA and IAS, bind your application to service instances of both types
```

File: java/security.md (hunk 4 above)
Comment (renejeglinsky) [reply java/security.md:56]:
```suggestion
   * only accept tokens issued by IAS --> bind your application to an [IAS service instance](https://help.sap.com/docs/IDENTITY_AUTHENTICATION)
```

File: java/security.md (hunk 3 above)
Comment (renejeglinsky) [reply java/security.md:55]:
```suggestion
   * only accept tokens issued by XSUAA --> bind your application to an [XSUAA service instance](../guides/authorization#xsuaa-configuration)
```

**Review by BraunMatthias — APPROVED** [review]

File: java/security.md [hunk 7]
```diff
@@ -35,59 +35,46 @@ Hence both, authentication and authorization, are essential for application secu
...
-1. Following dependencies are set:
-  * `resourceserver-security-spring-boot-starter` that brings Spring Security and [Java security library](https://github.com/SAP/cloud-security-xsuaa-integration)
-  * `cds-feature-identity`
-2. The application is bound to an [IAS service instance](https://help.sap.com/docs/IDENTITY_AUTHENTICATION)
-::: warning
-To enforce IAS authentication, make sure no XSUAA instance is bound to the CAP service at the same time.
-:::
-::: tip
-To allow forwarding to remote services, JWT tokens issued by IAS service do not contain authorization information. In particular, no scopes are included. Closing this gap is up to you in your application.
 :::
```
Comment (renejeglinsky) [inline java/security.md:55]: Restructured it, so that the bundles are mentioned first and individuals dependencies are in a collapsed section:
image

**Pull request merged**